### PR TITLE
Added views for Creating and updating sponsors from dashboard.

### LINKED ISSUE #727: As an owner or admin I can add new sponsors
@QuincyLarson @allella 

I think this needs to be part of the MVP, what do you reckon?

### PR DESCRIPTION
**Pull request by Ravichandra-C**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #727 

1. Added Sponsors tab to the dashboard view.
2. Created CreateSponsor and UpdateSponsor Mutations.
3. Created Create new sponsor and update sponsor form.
4. Created Dashboard datatable sponsor views.

### CODE REVIEW

File: client/src/modules/dashboard/Sponsors/components/SponsorForm.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import { useForm } from 'react-hook-form';
+import { Input } from '../../../../components/Form/Input';
+import { Sponsor, SponsorQuery } from 'generated/graphql';
+
+export type SponsorFormData = Omit<
+  Sponsor,
+  'id' | 'created_at' | 'updated_at' | 'events'
+>;
+
+interface SpornsorFormProps {
```
Comment (moT01) [inline client/src/modules/dashboard/Sponsors/components/SponsorForm.tsx:14]:
```suggestion
interface SponsorFormProps {
```

minor typo

File: client/src/modules/dashboard/Sponsors/components/SponsorForm.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    isRequired: true,
+  },
+  {
+    key: 'logo_path',
+    placeholder: 'www.freecodecamp.com',
+    label: 'Logo Path',
+    isRequired: true,
+  },
+];
+const SponsorForm: React.FC<SpornsorFormProps> = (props) => {
```
Comment (moT01) [inline client/src/modules/dashboard/Sponsors/components/SponsorForm.tsx:46]:
```suggestion
const SponsorForm: React.FC<SponsorFormProps> = (props) => {
```

File: client/src/modules/dashboard/Sponsors/graphql/queries.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      name
+      website
+      logo_path
+      type
+    }
+  }
+`;
+
+export const updateSponsor = gql`
+  mutation UpdateSponsor($data: UpdateSponsorInputs!, $updateSponsorId: Int!) {
```
Comment (moT01) [inline client/src/modules/dashboard/Sponsors/graphql/queries.ts:27]:
```suggestion
  mutation updateSponsor($data: UpdateSponsorInputs!, $updateSponsorId: Int!) {
```

Doesn't matter, but the rest of the query names seem to use camel case

File: client/src/modules/dashboard/Sponsors/pages/NewSponsorPage.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import { useCreateSponsorMutation } from 'generated/graphql';
+const NewSponsorPage: NextPage = () => {
+  const [loading, setLoading] = useState(false);
+  const router = useRouter();
+  const [createSponsor] = useCreateSponsorMutation({
+    refetchQueries: [{ query: Sponsors }],
+  });
+  const onSubmit = async (data: SponsorFormData) => {
+    setLoading(true);
+    console.log('Logging the data');
```
Comment (moT01) [inline client/src/modules/dashboard/Sponsors/pages/NewSponsorPage.tsx:16]:
```suggestion
```

Probably left this in accidentally?

File: client/src/modules/dashboard/Events/pages/EventPage.tsx [hunk 5]
```diff
@@ -90,11 +90,7 @@ export const EventPage: NextPage = () => {
           <h2>Venue: Online</h2>
         )}
       </Box>
-      {data.event.sponsors.length ? (
```
Comment (moT01) [inline client/src/modules/dashboard/Events/pages/EventPage.tsx:93]: I'm curious why this was removed? Do we not want to show the sponsors on the event page - or is there some other implementation in mind or something?

**Review by moT01 — COMMENTED** [review]
This looks real good @Ravichandra-C 🎉

There was a few minor things in the code I left suggestions on. My only real curiosity is why this removes the sponsors from being displayed on the event page.

File: client/src/modules/dashboard/Sponsors/components/SponsorForm.tsx (hunk 1 above)
Comment (Ravichandra-C) [reply client/src/modules/dashboard/Sponsors/components/SponsorForm.tsx:14]: @moT01 , I've fixed this.

File: client/src/modules/dashboard/Sponsors/components/SponsorForm.tsx (hunk 2 above)
Comment (Ravichandra-C) [reply client/src/modules/dashboard/Sponsors/components/SponsorForm.tsx:46]: @moT01 , I've fixed this.

File: client/src/modules/dashboard/Sponsors/graphql/queries.ts (hunk 3 above)
Comment (Ravichandra-C) [reply client/src/modules/dashboard/Sponsors/graphql/queries.ts:27]: @moT01 , I've fixed this.

File: client/src/modules/dashboard/Sponsors/pages/NewSponsorPage.tsx (hunk 4 above)
Comment (Ravichandra-C) [reply client/src/modules/dashboard/Sponsors/pages/NewSponsorPage.tsx:16]: @moT01 , I've fixed this.

File: client/src/modules/dashboard/Events/pages/EventPage.tsx (hunk 5 above)
Comment (Ravichandra-C) [reply client/src/modules/dashboard/Events/pages/EventPage.tsx:93]: @moT01, This removal is not intentional. This somehow got added to the staged files. Can I remove it  here?

Comment (Ravichandra-C) [reply client/src/modules/dashboard/Events/pages/EventPage.tsx]: @moT01 , I've added back the code.

**Comment by Ravichandra-C** [comment]
@moT01 , Thanks a lot for you feedback. I've fixed them. Could you please review it once again?

**Review by ojeytonwilliams — APPROVED** [review]
Nice work, @Ravichandra-C. LGTM :+1: 

If Tom's happy with the changes, we can get this in before https://github.com/freeCodeCamp/chapter/pull/804

**Review by moT01 — APPROVED** [review]
LGTM @Ravichandra-C 🎉

**Pull request merged**